main.go

```diff
@@ -569,42 +276,27 @@ func main() {
 			ReadTimeout:  readTimeout,  // best set per handler.
 			WriteTimeout: writeTimeout, // best set per handler.
 		}
-
 		g.Add(func() error {
-			level.Info(logger).Log("msg", "starting the HTTP server", "address", cfg.server.listen)
-
+			level.Info(tCfg.logger).Log("msg", "starting the HTTP server", "address", cfg.server.listen)
 			if tlsConfig != nil {
-				// serverCertFile and serverKeyFile passed in TLSConfig at initialization.
 				return s.ListenAndServeTLS("", "")
 			}
 
 			return s.ListenAndServe()
 		}, func(err error) {
 			// gracePeriod is duration the server gracefully shuts down.
-			const gracePeriod = gracePeriod
-
 			ctx, cancel := context.WithTimeout(context.Background(), gracePeriod)
 			defer cancel()
 
-			level.Info(logger).Log("msg", "shutting down the HTTP server")
+			level.Info(tCfg.logger).Log("msg", "shutting down the HTTP server")
 			_ = s.Shutdown(ctx)
 		})
 	}
 	{
-		h := internalserver.NewHandler(
-			internalserver.WithName("Internal - Observatorium API"),
-			internalserver.WithHealthchecks(healthchecks),
-			internalserver.WithPrometheusRegistry(reg),
-			internalserver.WithPProf(),
-		)
-
-		s := http.Server{
-			Addr:    cfg.server.listenInternal,
-			Handler: h,
-		}
-
+		healthchecks := commonHealthChecks(cfg, tCfg)
```

Comment:
possibly worth to inline (no need for single time used variable)

Reply:
this variable is used as func. parameter (setupInternalServer() )